pyscriptjs/src/pyexec.ts

```diff
@@ -0,0 +1,65 @@
+import { getLogger } from './logger';
+import { ensureUniqueId, addClasses } from './utils';
+import type { Runtime } from './runtime';
+
+const logger = getLogger('pyexec');
+
+export async function pyExec(runtime: Runtime, pysrc: string, outElem: HTMLElement)
+{
+    // this is the python function defined in pyscript.py
+    const set_current_display_target = runtime.globals.get('set_current_display_target');
+    ensureUniqueId(outElem);
+    set_current_display_target(outElem.id);
+    try {
+        try {
+            await runtime.run(pysrc);
+        }
+        catch (err) {
+            // XXX: currently we display exceptions in the same position as
+            // the output. But we probably need a better way to do that,
+            // e.g. allowing plugins to intercept exceptions and display them
+            // in a configurable way.
+            displayPyException(err, outElem);
+        }
+    }
+    finally {
+        set_current_display_target(undefined);
+    }
+}
+
+
+function displayPyException(err: any, errElem: HTMLElement) {
```

Comment:
Love this as a start - having a unified way to intercept and handle errors will be nice, but really good to have this behavior in one place.

Reply:
yes. The next step will be to have an unified place where to display errors when there is no other obvious place (e.g. if you raise an exception from an event handler).